centreon/www/front_src/src/CloudNotificationsConfiguration/Actions/DeleteButton.tsx

```diff
@@ -0,0 +1,178 @@
+import { useState } from 'react';
+
+import { useTranslation } from 'react-i18next';
+import { useQueryClient } from '@tanstack/react-query';
+import { makeStyles } from 'tss-react/mui';
+import {
+  includes,
+  isEmpty,
+  last,
+  length,
+  propEq,
+  split,
+  complement
+} from 'ramda';
+import { useAtomValue } from 'jotai';
+
+import { Box, alpha } from '@mui/material';
+import DeleteIcon from '@mui/icons-material/DeleteOutline';
+
+import {
+  IconButton,
+  ResponseError,
+  useMutationQuery,
+  useSnackbar,
+  ConfirmDialog,
+  Method
+} from '@centreon/ui';
+
+import {
+  labelDelete,
+  labelCancel,
+  labelDeleteNotification,
+  labelDeleteNotificationWarning,
+  labelFailedToDeleteNotifications
+} from '../translatedLabels';
+import { selectedRowsAtom } from '../atom';
+
+const useStyles = makeStyles()((theme) => ({
+  confirmButtons: {
+    '&:hover': {
+      background: alpha(theme.palette.error.main, 0.8)
+    },
+    background: theme.palette.error.main,
+    color: theme.palette.common.white
+  },
+  paper: {
+    width: theme.spacing(60)
+  }
+}));
+
+interface Props {
+  ariaLabel?: string;
+  className?: string;
+  disabled?: boolean;
+  fetchMethod: Method;
+  getEndpoint: () => string;
+  iconClassName?: string;
+  labelFailed: string;
+  labelSuccess: string;
+  notificationName?: string;
+  onSuccess?: () => void;
+  payload?: { ids: Array<number | string> };
+}
+
+const DeleteButton = ({
+  getEndpoint,
+  onSuccess,
+  labelSuccess,
+  labelFailed,
+  fetchMethod,
+  payload,
+  notificationName,
+  disabled = false,
+  iconClassName,
+  className,
+  ariaLabel
+}: Props): JSX.Element => {
+  const { classes } = useStyles();
+  const { t } = useTranslation();
+  const { showSuccessMessage, showErrorMessage, showWarningMessage } =
+    useSnackbar();
+
+  const [dialogOpen, setDialogOpen] = useState(false);
+  const selectedRows = useAtomValue(selectedRowsAtom);
+
+  const queryClient = useQueryClient();
+  const { isMutating, mutateAsync } = useMutationQuery({
+    defaultFailureMessage: t(labelFailed) as string,
+    getEndpoint,
+    method: fetchMethod
+  });
+
+  const onClick = (): void => setDialogOpen(true);
+  const onCancel = (): void => setDialogOpen(false);
+
+  const onConfirm = (): void => {
+    mutateAsync(payload || {}).then((response) => {
+      const {
+        isError,
+        statusCode,
+        message,
+        additionalInformation: data
+      } = response as ResponseError;
```

Comment:
_(see previous comment in customFetch)_

...and `additionalInformation` mapped back to `data` 😅